File: mobile/src/main/java/io/ashdavies/cinnamon/account/AccountActivity.java
package io.ashdavies.cinnamon.account;

import android.content.Intent;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.OnClick;
import io.ashdavies.cinnamon.R;
import io.ashdavies.cinnamon.activity.AbstractActivity;
import io.ashdavies.cinnamon.barcode.BarcodeOnboardingActivity;
import javax.inject.Inject;

public class AccountActivity extends AbstractActivity implements AccountView {

  private static final int REQUEST_BARCODE_CAPTURE = 0x14;

  @BindView(R.id.headline_welcome) TextView welcome;

  @Inject AccountPresenter presenter;
  @Inject AccountActivity activity;

  @Override
  protected int getLayoutId() {
    return R.layout.activity_account;
  }

  @Override
  protected int getMenuId() {
    return R.menu.activity_empty;
  }

  @OnClick(R.id.action_account_google)
  void onGoogleAccountClick() {
  }

  @OnClick(R.id.action_account_kraken)
  void onKrakenAccountClick() {
    startActivityForResult(new Intent(this, BarcodeOnboardingActivity.class), REQUEST_BARCODE_CAPTURE);
  }

  @Override
  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
    switch (requestCode) {
      case REQUEST_BARCODE_CAPTURE:
        presenter.onBarcodeResult(resultCode, data);
        return;

      default:
        super.onActivityResult(requestCode, resultCode, data);
    }
  }
}

File: mobile/src/main/java/io/ashdavies/cinnamon/account/AccountActivity.kt
package io.ashdavies.cinnamon.account

import android.content.Intent
import android.widget.TextView
import butterknife.BindView
import butterknife.OnClick
import io.ashdavies.cinnamon.R
import io.ashdavies.cinnamon.activity.AbstractActivity
import io.ashdavies.cinnamon.barcode.BarcodeOnboardingActivity
import javax.inject.Inject

class AccountActivity : AbstractActivity(), AccountView {

  @BindView(R.id.headline_welcome) internal lateinit var welcome: TextView

  @Inject internal lateinit var presenter: AccountPresenter
  @Inject internal lateinit var activity: AccountActivity

  override fun getLayoutId(): Int {
    return R.layout.activity_account
  }

  override fun getMenuId(): Int {
    return R.menu.activity_empty
  }

  @OnClick(R.id.action_account_google)
  internal fun onGoogleAccountClick() {
    presenter.onGoogleLogin()
  }

  @OnClick(R.id.action_account_kraken)
  internal fun onKrakenAccountClick() {
    startActivityForResult(Intent(this, BarcodeOnboardingActivity::class.java), REQUEST_BARCODE_CAPTURE)
  }

  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
    when (requestCode) {
      REQUEST_BARCODE_CAPTURE -> {
        presenter.onBarcodeResult(resultCode, data)
        return
      }

      else -> super.onActivityResult(requestCode, resultCode, data)
    }
  }

  companion object {

    private val REQUEST_BARCODE_CAPTURE = 0x14
  }
}
